File: src/constants/jobTypes.js
// Job types constants - single source of truth
// These must match the backend constants exactly

export const JOB_TYPES = [
  { code: "TRAVELLER_EVICTION",          label: "Traveller Eviction" },
  { code: "SQUATTER_EVICTION",           label: "Squatter Eviction" },
  { code: "TRAVELLER_NOTICE_SERVE",      label: "Traveller Notice Serve" },
  { code: "SQUATTER_NOTICE_SERVE",       label: "Squatter Notice Serve" },
  { code: "VEHICLE_TORTS_NOTICE",        label: "Vehicle Torts Notice" },
  { code: "LEASE_FORFEITURE",            label: "Lease Forfeiture" },
  { code: "ROUGH_SLEEPER",               label: "Rough sleeper" },
]

// Helper functions
export const getJobTypeLabel = (code) => {
  const jobType = JOB_TYPES.find(jt => jt.code === code)
  return jobType ? jobType.label : code || "Unknown"
}

export const isValidJobTypeCode = (code) => {
  return JOB_TYPES.some(jt => jt.code === code)
}

// Export codes for validation
export const JOB_TYPE_CODES = JOB_TYPES.map(jt => jt.code)

// Backward compatibility - keep the old format for existing code
export const getJobTypeLabel_Legacy = (value) => {
  // First try new format
  const newJobType = JOB_TYPES.find(type => type.code === value)
  if (newJobType) return newJobType.label

  // Fallback to returning the value itself for old job types
  return value || "Unknown"
}

File: src/constants/job_types.py
# Job types constants - single source of truth
# These must match the frontend constants exactly

JOB_TYPES = {
    "TRAVELLER_EVICTION": "Traveller Eviction",
    "SQUATTER_EVICTION": "Squatter Eviction",
    "TRAVELLER_NOTICE_SERVE": "Traveller Notice Serve",
    "SQUATTER_NOTICE_SERVE": "Squatter Notice Serve",
    "VEHICLE_TORTS_NOTICE": "Vehicle Torts Notice",
    "LEASE_FORFEITURE": "Lease Forfeiture",
    "ROUGH_SLEEPER": "Rough sleeper",
}

ALLOWED_JOB_TYPE_CODES = set(JOB_TYPES.keys())

def get_job_type_label(code):
    """Get human-readable label for a job type code"""
    return JOB_TYPES.get(code, code or "Unknown")

def is_valid_job_type_code(code):
    """Check if a job type code is valid"""
    return code in ALLOWED_JOB_TYPE_CODES